addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -409,7 +409,12 @@ export const editorCommands = {
         const changedElements = [];
         const defaultDirection = editor.options.direction;
         const shouldApplyStyle = !isSelectionFormat(editor.editable, 'switchDirection');
-        for (const block of new Set(selectedTextNodes.map(textNode => closestElement(textNode, 'ul,ol') || closestBlock(textNode)))) {
+        const blocks = new Set(selectedTextNodes.map(textNode => closestElement(textNode, 'ul,ol') || closestBlock(textNode)))
+        blocks.forEach(block => {
+            const nestedUL = Array.from(block.getElementsByTagName('UL'));
+            nestedUL.forEach(ul => blocks.add(ul));
```

Comment:
1. Why check nested UL and not OL ?
2. This seems over complicated. I think something like this should work :

```javascript
blocks = [...blocks, ...block.querySelectorAll('ul,ol')]
```

Reply:
@sebgeelen As checklists are Unordered-list, I considered 'ul' only.